text/00xx-smart-contracts-coretime-chain.md

```diff
@@ -0,0 +1,102 @@
+# RFC-0002: Smart Contracts on the Coretime Chain
+
+|                 |                                                                                             |
+| --------------- | ------------------------------------------------------------------------------------------- |
+| **Start Date**  | 2024-06-09                                                                                  |
+| **Description** | Implement smart contracts on the Coretime chain                                             |
+| **Authors**     | [Aurora Poppyseed](https://github.com/poppyseedDev/), [Phil Lucksok](https://github.com/phillux)  |
+
+## Summary
+
+This RFC proposes the implementation of smart contracts on the Coretime chain to enhance flexibility and enable complex decentralized applications, including secondary market functionalities.
+
+## Motivation
+
+Currently, the Coretime chain lacks the capability to support smart contracts, which limits the range of decentralized applications that can be developed and deployed. By enabling smart contracts, the Coretime chain can facilitate more sophisticated functionalities such as automated region trading, dynamic pricing mechanisms, and other decentralized applications that require programmable logic. This will enhance the utility of the Coretime chain, attract more developers, and create more opportunities for innovation.
+
+Additionally, while there is a proposal (#885) to allow EVM-compatible contracts on Polkadot’s Asset Hub, the implementation of smart contracts directly on the Coretime chain will provide synchronous interactions and avoid the complexities of asynchronous operations via XCM. 
+
+## Stakeholders
+
+Primary stakeholders include:
+- Developers working on the Coretime chain.
+- Users who want to deploy decentralized applications on the Coretime chain.
+- Community members interested in expanding the capabilities of the Coretime chain.
+- Secondary Coretime marketplaces.
+
+## Explanation
+
+This RFC introduces the following key components:
+
+1. **Smart Contract Support**:
+    - Implement support for deploying and executing smart contracts on the Coretime chain.
+    - Use a well-established smart contract platform, such as Ethereum’s Solidity or Polkadot's Ink!, to ensure compatibility and ease of use.
```

Comment:
Solidity and ink aren't platforms, maybe you mean, "Use a well-supported smart contract virtual machine that supports contracts written in either Solidity or ink."?
```suggestion
    - Integrate support for deploying and executing smart contracts on the Coretime chain.
    - Use a well-established smart contract platform, such as Ethereum’s Solidity or Polkadot's Ink!, to ensure compatibility and ease of use.
```